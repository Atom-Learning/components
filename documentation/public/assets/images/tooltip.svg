<svg width="153" height="48" fill="none" xmlns="http://www.w3.org/2000/svg"><g filter="url(#a)"><path d="m76.6 2 4.9 5h-10l5.1-5Z" fill="#333"/><rect x="3" y="7" width="147" height="36" rx="4" fill="#333"/><path d="M19.77 20.61v-1.25h8.73v1.25h-3.66V31h-1.4V20.61h-3.67Zm12.06 5.14V31H30.5V19.36h1.34v4.28h.12c.2-.45.5-.81.92-1.08.41-.27.96-.4 1.65-.4.59 0 1.11.12 1.56.36.45.23.8.6 1.04 1.08.25.49.37 1.1.37 1.85V31h-1.34v-5.45c0-.7-.18-1.23-.54-1.61a1.93 1.93 0 0 0-1.48-.58c-.44 0-.83.1-1.18.28-.35.19-.62.46-.82.81-.2.36-.3.8-.3 1.3ZM39.95 31v-8.73h1.34V31h-1.34Zm.68-10.18a.95.95 0 0 1-.68-.27.86.86 0 0 1-.28-.64c0-.25.1-.46.28-.64a.95.95 0 0 1 .68-.27c.26 0 .48.09.67.27.19.18.28.39.28.64s-.1.46-.28.64a.93.93 0 0 1-.67.27Zm9.3 3.4-1.21.35c-.08-.2-.19-.4-.34-.59a1.56 1.56 0 0 0-.59-.47 2.15 2.15 0 0 0-.96-.2c-.53 0-.98.13-1.33.38-.35.24-.53.55-.53.92 0 .34.12.6.36.8.25.19.63.35 1.14.48l1.3.31c.78.2 1.36.48 1.74.87.38.39.57.89.57 1.5 0 .5-.14.95-.43 1.34-.28.4-.68.7-1.2.93-.5.23-1.1.34-1.78.34-.89 0-1.62-.2-2.2-.58a2.57 2.57 0 0 1-1.11-1.7l1.27-.3c.12.46.35.81.69 1.05.34.23.78.35 1.33.35a2.5 2.5 0 0 0 1.5-.4c.36-.27.55-.59.55-.96 0-.3-.1-.56-.32-.77-.21-.2-.54-.36-.98-.46l-1.45-.34a3.47 3.47 0 0 1-1.76-.88c-.38-.4-.56-.9-.56-1.5 0-.5.14-.94.41-1.32a2.9 2.9 0 0 1 1.15-.89 4 4 0 0 1 1.64-.32c.87 0 1.55.19 2.04.57.5.38.85.88 1.05 1.5ZM56.6 31v-8.73h1.34V31H56.6Zm.68-10.18a.95.95 0 0 1-.67-.27.86.86 0 0 1-.28-.64c0-.25.1-.46.28-.64a.95.95 0 0 1 .67-.27.9.9 0 0 1 .67.27.85.85 0 0 1 0 1.28.93.93 0 0 1-.67.27Zm9.3 3.4-1.2.35c-.08-.2-.2-.4-.34-.59a1.56 1.56 0 0 0-.59-.47 2.15 2.15 0 0 0-.96-.2c-.53 0-.98.13-1.34.38-.35.24-.52.55-.52.92 0 .34.12.6.36.8.24.19.62.35 1.14.48l1.3.31c.77.2 1.35.48 1.74.87.38.39.57.89.57 1.5 0 .5-.14.95-.43 1.34-.29.4-.68.7-1.2.93-.5.23-1.1.34-1.78.34a3.9 3.9 0 0 1-2.2-.58A2.57 2.57 0 0 1 60 28.9l1.28-.3c.12.46.35.81.68 1.05.34.23.79.35 1.34.35a2.5 2.5 0 0 0 1.49-.4c.37-.27.55-.59.55-.96 0-.3-.1-.56-.31-.77-.22-.2-.54-.36-.98-.46l-1.46-.34a3.47 3.47 0 0 1-1.76-.88c-.37-.4-.55-.9-.55-1.5 0-.5.13-.94.41-1.32.28-.37.66-.67 1.14-.89a4 4 0 0 1 1.65-.32c.86 0 1.54.19 2.03.57.5.38.85.88 1.06 1.5Zm9.25 6.98c-.56 0-1.06-.1-1.5-.3a2.56 2.56 0 0 1-1.08-.92c-.27-.4-.4-.89-.4-1.46 0-.5.1-.9.3-1.21.2-.32.46-.56.78-.74.33-.18.7-.31 1.1-.4.4-.1.8-.16 1.2-.22l1.3-.15c.33-.04.57-.1.72-.19.16-.08.24-.23.24-.45v-.05c0-.56-.16-1-.46-1.3-.3-.31-.77-.47-1.39-.47a2.39 2.39 0 0 0-2.27 1.32l-1.27-.46a3.06 3.06 0 0 1 2.16-1.86 5.19 5.19 0 0 1 2.3-.08c.37.07.73.2 1.07.4.34.2.63.52.86.93.22.41.34.97.34 1.66V31h-1.34v-1.18h-.07a2.62 2.62 0 0 1-1.3 1.16c-.36.15-.79.22-1.3.22Zm.2-1.2c.53 0 .98-.1 1.34-.31a2.13 2.13 0 0 0 1.12-1.85v-1.23a.99.99 0 0 1-.38.2c-.19.04-.4.1-.66.13a29.81 29.81 0 0 1-1.28.17c-.35.05-.68.12-.98.23-.3.1-.54.24-.73.44-.18.2-.27.47-.27.81 0 .47.17.82.52 1.06.35.23.79.35 1.32.35Zm14.54-7.73v1.14h-4.52v-1.14h4.52Zm-3.2-2.09h1.34v8.32c0 .38.05.66.16.85.11.19.26.31.43.38.18.06.37.09.57.09.14 0 .26 0 .36-.02l.23-.05.27 1.2a2.73 2.73 0 0 1-1 .16 2.7 2.7 0 0 1-1.11-.24 2.21 2.21 0 0 1-.9-.75 2.14 2.14 0 0 1-.35-1.26v-8.68Zm8.67 11a3.8 3.8 0 0 1-2.08-.56 3.84 3.84 0 0 1-1.38-1.57 5.32 5.32 0 0 1-.5-2.37c0-.9.17-1.7.5-2.38.33-.68.8-1.2 1.38-1.58a3.8 3.8 0 0 1 2.08-.56c.79 0 1.48.19 2.07.56.6.38 1.05.9 1.38 1.58.34.68.5 1.47.5 2.38 0 .9-.16 1.7-.5 2.37-.33.67-.79 1.2-1.38 1.57-.6.37-1.28.56-2.07.56Zm0-1.2c.6 0 1.09-.16 1.48-.46.38-.31.67-.71.85-1.21a4.8 4.8 0 0 0 0-3.26c-.18-.5-.47-.91-.85-1.22a2.28 2.28 0 0 0-1.48-.47c-.6 0-1.1.16-1.48.47-.39.31-.67.72-.86 1.22a4.8 4.8 0 0 0 0 3.26c.19.5.47.9.86 1.2.39.31.88.47 1.48.47Zm9.54 1.2a3.8 3.8 0 0 1-2.07-.56 3.93 3.93 0 0 1-1.39-1.57 5.32 5.32 0 0 1-.49-2.37c0-.9.17-1.7.5-2.38a3.8 3.8 0 0 1 1.38-1.58 3.8 3.8 0 0 1 2.07-.56c.8 0 1.48.19 2.07.56.6.38 1.06.9 1.39 1.58.33.68.5 1.47.5 2.38 0 .9-.17 1.7-.5 2.37a3.8 3.8 0 0 1-1.39 1.57c-.59.37-1.28.56-2.07.56Zm0-1.2c.6 0 1.1-.16 1.48-.46.39-.31.67-.71.86-1.21a4.8 4.8 0 0 0 0-3.26c-.19-.5-.47-.91-.86-1.22a2.28 2.28 0 0 0-1.48-.47c-.6 0-1.09.16-1.47.47-.39.31-.67.72-.86 1.22a4.77 4.77 0 0 0 0 3.26c.19.5.47.9.86 1.2.38.31.88.47 1.47.47Zm7.35-10.62V31h-1.34V19.36h1.34Zm6.25 2.91v1.14h-4.52v-1.14h4.52Zm-3.2-2.09h1.33v8.32c0 .38.06.66.17.85.11.19.26.31.43.38.18.06.37.09.56.09.15 0 .27 0 .37-.02l.22-.05.28 1.2a2.73 2.73 0 0 1-1 .16 2.7 2.7 0 0 1-1.12-.24 2.21 2.21 0 0 1-.9-.75 2.14 2.14 0 0 1-.35-1.26v-8.68ZM121.2 31v-8.73h1.34V31h-1.34Zm.68-10.18a.95.95 0 0 1-.68-.27.86.86 0 0 1-.28-.64c0-.25.1-.46.28-.64a.95.95 0 0 1 .68-.27c.26 0 .48.09.67.27.19.18.28.39.28.64s-.1.46-.28.64a.93.93 0 0 1-.67.27Zm3.11 13.45v-12h1.3v1.39h.16l.4-.58c.19-.24.44-.45.77-.64s.78-.28 1.35-.28a3.46 3.46 0 0 1 3.26 2.11c.31.68.47 1.47.47 2.39 0 .92-.16 1.72-.47 2.4a3.42 3.42 0 0 1-3.23 2.12 2.8 2.8 0 0 1-1.35-.28 2.5 2.5 0 0 1-.78-.64 8.77 8.77 0 0 1-.42-.6h-.12v4.61H125Zm1.32-7.63c0 .66.1 1.24.3 1.74.18.5.47.9.84 1.18.37.28.82.42 1.36.42a2.2 2.2 0 0 0 1.4-.45c.38-.3.67-.7.86-1.2.19-.5.29-1.07.29-1.7 0-.6-.1-1.16-.29-1.65-.18-.5-.47-.9-.84-1.18a2.2 2.2 0 0 0-1.42-.44 2.24 2.24 0 0 0-2.22 1.57 4.73 4.73 0 0 0-.28 1.7Z" fill="#fff"/></g><defs><filter id="a" x="0" y="0" width="153" height="48" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"/><feColorMatrix in="SourceAlpha" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/><feOffset dy="2"/><feGaussianBlur stdDeviation="1.5"/><feColorMatrix values="0 0 0 0 0.2 0 0 0 0 0.2 0 0 0 0 0.2 0 0 0 0.15 0"/><feBlend in2="BackgroundImageFix" result="effect1_dropShadow_821_54474"/><feColorMatrix in="SourceAlpha" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/><feOffset dy="1"/><feGaussianBlur stdDeviation="1.5"/><feColorMatrix values="0 0 0 0 0.2 0 0 0 0 0.2 0 0 0 0 0.2 0 0 0 0.1 0"/><feBlend in2="effect1_dropShadow_821_54474" result="effect2_dropShadow_821_54474"/><feBlend in="SourceGraphic" in2="effect2_dropShadow_821_54474" result="shape"/></filter></defs></svg>