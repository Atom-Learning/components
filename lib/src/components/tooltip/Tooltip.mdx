---
title: Tooltip
component: Tooltip,Tooltip.Trigger,Tooltip.Content,Tooltip.Provider, Tooltip.Portal
description: A component that displays information related to an element when the element receives keyboard focus or the mouse hovers over it
category: Surfaces
---

`Tooltip` exports 4 components that combine to create our tooltip. The `Tooltip.Trigger` requires a React component child (or passed to `as`) to allow the necessary event bindings and accessible properties to be cloned.

Also, note that the component must accept a `ref`.

`Tooltip.Provider` is **required to avoid regressions** and should be present at the top level of the app. 

Read more about the underlying UI component on the [Radix UI documentation site](https://radix-ui.com/primitives/docs/components/tooltip).

```tsx preview
<Tooltip.Provider>
  <Tooltip>
    <Tooltip.Trigger>
      <Text>Hover on me</Text>
    </Tooltip.Trigger>
    <Tooltip.Content>This is the tooltip content, hello</Tooltip.Content>
  </Tooltip>
</Tooltip.Provider>
```
