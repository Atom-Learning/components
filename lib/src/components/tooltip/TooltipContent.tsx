import { Arrow, Content } from '@radix-ui/react-tooltip'
import * as React from 'react'

import { styled } from '~/stitches'
import {
  slideDownAndFade,
  slideLeftAndFade,
  slideRightAndFade,
  slideUpAndFade
} from '~/utilities'

const StyledContent = styled(Content, {
  backgroundColor: '$tonal500',
  borderRadius: '$0',
  boxShadow: '$0',
  color: 'white',
  fontFamily: '$body',
  fontSize: '$sm',
  lineHeight: 1.5,
  px: '$3',
  py: '$2',
  '@allowMotion': {
    animationDuration: '75ms',
    animationTimingFunction: 'cubic-bezier(0.16, 1, 0.3, 1)',
    willChange: 'transform, opacity',
    '&[data-state="delayed-open"]': {
      '&[data-side="top"]': { animationName: slideDownAndFade },
      '&[data-side="right"]': { animationName: slideLeftAndFade },
      '&[data-side="bottom"]': { animationName: slideUpAndFade },
      '&[data-side="left"]': { animationName: slideRightAndFade }
    }
  },
  variants: {
    size: {
      sm: { maxWidth: '100px' },
      md: { maxWidth: '250px' },
      lg: { maxWidth: '400px' }
    }
  }
})

const StyledArrow = styled(Arrow, {
  fill: '$tonal500',
  '[data-align="end"] &': { mr: '$2' },
  '[data-align="start"] &': { ml: '$2' }
})

type TooltipContentProps = React.ComponentProps<typeof StyledContent> &
  React.ComponentProps<typeof Content>

export const TooltipContent: React.FC<TooltipContentProps> = ({
  children,
  side = 'top',
  sideOffset = 4,
  size = 'md',
  ...remainingProps
}) => (
  <StyledContent
    side={side}
    sideOffset={sideOffset}
    size={size}
    {...remainingProps}
  >
    {children}
    <StyledArrow />
  </StyledContent>
)
