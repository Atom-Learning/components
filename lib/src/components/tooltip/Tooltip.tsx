import { Portal, Provider, Root, Trigger } from '@radix-ui/react-tooltip'
import * as React from 'react'

import { styled } from '~/stitches'

import { TooltipContent } from './TooltipContent'

type TooltipProps = React.ComponentProps<typeof Root>

export const Tooltip: React.FC<TooltipProps> & {
  Content: typeof TooltipContent
  Portal: typeof Portal
  Trigger: typeof Trigger
  Provider: typeof Provider
} = ({ children, delayDuration = 350, ...remainingProps }) => (
  <Root delayDuration={delayDuration} {...remainingProps}>
    {children}
  </Root>
)

Tooltip.Content = TooltipContent
Tooltip.Trigger = styled(Trigger, {})
Tooltip.Portal = Portal
Tooltip.Provider = Provider

Tooltip.displayName = 'Tooltip'
