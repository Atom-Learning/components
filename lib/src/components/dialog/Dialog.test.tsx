import type { RenderResult } from '@testing-library/react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Tooltip } from '../tooltip'
import { Dialog } from '.'

const DialogTest = (props) => (
  <Tooltip.Provider>
    <Dialog {...props}>
      <Dialog.Trigger>TRIGGER</Dialog.Trigger>
      <Dialog.Content>
        <Dialog.Heading>TITLE</Dialog.Heading>
        CONTENT
        <Dialog.Footer>FOOTER</Dialog.Footer>
      </Dialog.Content>
    </Dialog>
  </Tooltip.Provider>
)

describe(`Dialog component`, () => {
  let rendered: RenderResult
  let trigger

  beforeEach(() => {
    rendered = render(<DialogTest />)
    trigger = rendered.getByText('TRIGGER')
  })

  it('renders the trigger with the popover hidden by default', async () => {
    expect(await screen.queryByText('CONTENT')).not.toBeInTheDocument()
    expect(rendered.container).toMatchSnapshot()
  })

  it('opens the popover once trigger is clicked', async () => {
    render(<DialogTest />)

    expect(await trigger).toBeInTheDocument()
    expect(await screen.queryByText('CONTENT')).not.toBeInTheDocument()

    trigger.focus()
    await userEvent.click(trigger)

    expect(await screen.queryByText('CONTENT')).toBeInTheDocument()

    const dialog = await screen.getByRole('dialog')
    expect(dialog).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    expect(await axe(rendered.container)).toHaveNoViolations()
  })

  describe(`when open`, () => {
    beforeEach(async () => {
      await userEvent.click(trigger)
    })

    it('has no programmatically detectable a11y issues', async () => {
      expect(await axe(rendered.container)).toHaveNoViolations()
    })
  })
})

describe('Dialog component without close button', () => {
  it('renders', async () => {
    await render(
      <Tooltip.Provider>
        {' '}
        <Dialog>
          <Dialog.Trigger>TRIGGER</Dialog.Trigger>
          <Dialog.Content showCloseButton={false}>CONTENT</Dialog.Content>
        </Dialog>
      </Tooltip.Provider>
    )

    const trigger = await screen.getByText('TRIGGER')
    await userEvent.click(trigger)

    const dialog = await screen.getByRole('dialog')
    expect(dialog).toMatchSnapshot()
  })
})

describe('Dialog component with custom background', () => {
  it('renders', async () => {
    await render(
      <Tooltip.Provider>
        {' '}
        <Dialog>
          <Dialog.Trigger>TRIGGER</Dialog.Trigger>
          <Dialog.Content>
            <Dialog.Background>CUSTOM BACKGROUND</Dialog.Background>
            CONTENT
          </Dialog.Content>
        </Dialog>
      </Tooltip.Provider>
    )

    const trigger = await screen.getByText('TRIGGER')
    await userEvent.click(trigger)

    const dialog = await screen.getByRole('dialog')

    // Since background is a sibling to content
    expect(dialog.parentElement).toMatchSnapshot()
  })
})
