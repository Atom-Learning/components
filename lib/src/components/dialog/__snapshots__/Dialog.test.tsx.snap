// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component opens the popover once trigger is clicked 1`] = `
@media  {
  .c-oUmPa {
    align-items: center;
    -webkit-appearance: none;
    appearance: none;
    background: white;
    border: unset;
    border-radius: var(--radii-0);
    box-sizing: border-box;
    cursor: pointer;
    display: flex;
    flex-shrink: 0;
    justify-content: center;
    padding: unset;
    transition: all 100ms ease-out;
  }

  .c-oUmPa[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-jryFOR {
    font-family: var(--fonts-display);
    font-weight: 700;
    margin: 0;
  }

  .c-dhzjXW {
    display: flex;
  }

  .c-dIzwvf {
    background: white;
    box-shadow: var(--shadows-3);
    box-sizing: border-box;
    left: 50%;
    max-width: 90vw;
    max-height: 90vh;
    overflow-y: auto;
    padding: var(--space-5);
    position: fixed;
    top: 50%;
    transform: translate3d(-50%, -50%, 0);
    z-index: 1147483646;
  }

  .c-dIzwvf:focus {
    outline: none;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-dIzwvf[data-state="open"] {
      animation: k-iaWRFC 550ms cubic-bezier(0.22, 1, 0.36, 1);
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-dIzwvf[data-state="closed"] {
      animation: k-gturUW 550ms cubic-bezier(0.22, 1, 0.36, 1);
    }
}

  .c-fWhrtB {
    background-color: var(--colors-alpha600);
    position: fixed;
    inset: 0;
    z-index: 1147483646;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-fWhrtB[data-state="open"] {
      animation: k-eyOShd 250ms ease-out;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-fWhrtB[data-state="closed"] {
      animation: k-bHbNKp 550ms ease-out;
    }
}
}

@media  {
  .c-oUmPa-PrXKS-size-md {
    height: var(--sizes-4);
    width: var(--sizes-4);
  }

  .c-hMRxhp-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-jryFOR-dtbHih-size-sm {
    font-size: var(--fontSizes-lg);
    line-height: 1.14;
  }

  .c-jryFOR-dtbHih-size-sm::before {
    content: '';
    margin-bottom: -0.2174em;
    display: table;
  }

  .c-jryFOR-dtbHih-size-sm::after {
    content: '';
    margin-top: -0.2254em;
    display: table;
  }

  .c-dhzjXW-hmJoPD-justify-end {
    justify-content: end;
  }

  .c-dhzjXW-dvnNgW-gap-3 {
    gap: var(--space-3);
  }

  .c-dIzwvf-cJSEWp-size-sm {
    width: 480px;
    height: auto;
    max-width: auto;
    max-height: auto;
  }

@supports (height: 100svh) {
    .c-dIzwvf-cJSEWp-size-sm {
      height: auto;
      max-height: auto;
    }
}

  .c-dIzwvf-cJSEWp-size-sm {
    border-radius: var(--radii-1);
  }
}

@media  {
  .c-oUmPa-jCxibG-cv {
    background: transparent;
    color: var(--colors-tonal300);
  }

  .c-oUmPa-jCxibG-cv:not(:disabled):hover,
  .c-oUmPa-jCxibG-cv:not(:disabled):focus {
    color: var(--colors-primary900);
  }

  .c-oUmPa-jCxibG-cv:not(:disabled):active {
    color: var(--colors-primary1000);
  }
}

@media  {
  .c-oUmPa-ifOCxRY-css {
    position: absolute;
    right: var(--space-4);
    top: var(--space-4);
    height: var(--sizes-5);
    width: var(--sizes-5);
  }

  .c-jryFOR-ijkrzPY-css {
    border-bottom: 1px solid var(--colors-tonal100);
    font-family: var(--fonts-body);
    padding: 0 var(--space-5) var(--space-5) var(--space-5);
    margin-bottom: var(--space-5);
    margin-left: calc(var(--space-5)*-1);
    margin-right: calc(var(--space-5)*-1);
    box-sizing: content-box;
  }

  .c-dhzjXW-illFGmD-css {
    padding-top: var(--space-4);
  }
}

<div
  aria-describedby="radix-:r5:"
  aria-label="Dialog"
  aria-labelledby="radix-:r4:"
  class="c-dIzwvf c-dIzwvf-cJSEWp-size-sm"
  data-state="open"
  id="radix-:r3:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <button
    aria-label="Close dialog"
    class="c-oUmPa c-oUmPa-PrXKS-size-md c-oUmPa-jCxibG-cv c-oUmPa-ifOCxRY-css"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="c-hMRxhp c-hMRxhp-dMrjnF-size-md c-hMRxhp-iPJLV-css"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6.343 6.343 11.314 11.314m-11.314 0L17.657 6.343"
      />
    </svg>
  </button>
  <h2
    class="c-jryFOR c-jryFOR-dtbHih-size-sm c-jryFOR-ijkrzPY-css"
  >
    TITLE
  </h2>
  CONTENT
  <div
    class="c-dhzjXW c-dhzjXW-hmJoPD-justify-end c-dhzjXW-dvnNgW-gap-3 c-dhzjXW-illFGmD-css"
  >
    FOOTER
  </div>
</div>
`;

exports[`Dialog component renders the trigger with the popover hidden by default 1`] = `
<div>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    TRIGGER
  </button>
</div>
`;

exports[`Dialog component with custom background renders 1`] = `
@media  {
  .c-oUmPa {
    align-items: center;
    -webkit-appearance: none;
    appearance: none;
    background: white;
    border: unset;
    border-radius: var(--radii-0);
    box-sizing: border-box;
    cursor: pointer;
    display: flex;
    flex-shrink: 0;
    justify-content: center;
    padding: unset;
    transition: all 100ms ease-out;
  }

  .c-oUmPa[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-jryFOR {
    font-family: var(--fonts-display);
    font-weight: 700;
    margin: 0;
  }

  .c-dhzjXW {
    display: flex;
  }

  .c-dIzwvf {
    background: white;
    box-shadow: var(--shadows-3);
    box-sizing: border-box;
    left: 50%;
    max-width: 90vw;
    max-height: 90vh;
    overflow-y: auto;
    padding: var(--space-5);
    position: fixed;
    top: 50%;
    transform: translate3d(-50%, -50%, 0);
    z-index: 1147483646;
  }

  .c-dIzwvf:focus {
    outline: none;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-dIzwvf[data-state="open"] {
      animation: k-iaWRFC 550ms cubic-bezier(0.22, 1, 0.36, 1);
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-dIzwvf[data-state="closed"] {
      animation: k-gturUW 550ms cubic-bezier(0.22, 1, 0.36, 1);
    }
}

  .c-fWhrtB {
    background-color: var(--colors-alpha600);
    position: fixed;
    inset: 0;
    z-index: 1147483646;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-fWhrtB[data-state="open"] {
      animation: k-eyOShd 250ms ease-out;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-fWhrtB[data-state="closed"] {
      animation: k-bHbNKp 550ms ease-out;
    }
}

  .c-kNieLL {
    width: 100vw;
    height: 100vh;
  }
}

@media  {
  .c-oUmPa-PrXKS-size-md {
    height: var(--sizes-4);
    width: var(--sizes-4);
  }

  .c-hMRxhp-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-jryFOR-dtbHih-size-sm {
    font-size: var(--fontSizes-lg);
    line-height: 1.14;
  }

  .c-jryFOR-dtbHih-size-sm::before {
    content: '';
    margin-bottom: -0.2174em;
    display: table;
  }

  .c-jryFOR-dtbHih-size-sm::after {
    content: '';
    margin-top: -0.2254em;
    display: table;
  }

  .c-dhzjXW-hmJoPD-justify-end {
    justify-content: end;
  }

  .c-dhzjXW-dvnNgW-gap-3 {
    gap: var(--space-3);
  }

  .c-dIzwvf-cJSEWp-size-sm {
    width: 480px;
    height: auto;
    max-width: auto;
    max-height: auto;
  }

@supports (height: 100svh) {
    .c-dIzwvf-cJSEWp-size-sm {
      height: auto;
      max-height: auto;
    }
}

  .c-dIzwvf-cJSEWp-size-sm {
    border-radius: var(--radii-1);
  }
}

@media  {
  .c-oUmPa-jCxibG-cv {
    background: transparent;
    color: var(--colors-tonal300);
  }

  .c-oUmPa-jCxibG-cv:not(:disabled):hover,
  .c-oUmPa-jCxibG-cv:not(:disabled):focus {
    color: var(--colors-primary900);
  }

  .c-oUmPa-jCxibG-cv:not(:disabled):active {
    color: var(--colors-primary1000);
  }
}

@media  {
  .c-oUmPa-ifOCxRY-css {
    position: absolute;
    right: var(--space-4);
    top: var(--space-4);
    height: var(--sizes-5);
    width: var(--sizes-5);
  }

  .c-jryFOR-ijkrzPY-css {
    border-bottom: 1px solid var(--colors-tonal100);
    font-family: var(--fonts-body);
    padding: 0 var(--space-5) var(--space-5) var(--space-5);
    margin-bottom: var(--space-5);
    margin-left: calc(var(--space-5)*-1);
    margin-right: calc(var(--space-5)*-1);
    box-sizing: content-box;
  }

  .c-dhzjXW-illFGmD-css {
    padding-top: var(--space-4);
  }
}

<div
  class="c-fWhrtB"
  data-state="open"
  id="modal_overlay"
  style="pointer-events: auto;"
>
  <div
    aria-hidden="true"
    class="c-PJLV c-kNieLL"
    data-aria-hidden="true"
  >
    CUSTOM BACKGROUND
  </div>
  <div
    aria-describedby="radix-:rk:"
    aria-label="Dialog"
    aria-labelledby="radix-:rj:"
    class="c-dIzwvf c-dIzwvf-cJSEWp-size-sm"
    data-state="open"
    id="radix-:ri:"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
  >
    <button
      aria-label="Close dialog"
      class="c-oUmPa c-oUmPa-PrXKS-size-md c-oUmPa-jCxibG-cv c-oUmPa-ifOCxRY-css"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c-hMRxhp c-hMRxhp-dMrjnF-size-md c-hMRxhp-iPJLV-css"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m6.343 6.343 11.314 11.314m-11.314 0L17.657 6.343"
        />
      </svg>
    </button>
    CONTENT
  </div>
</div>
`;

exports[`Dialog component without close button renders 1`] = `
@media  {
  .c-oUmPa {
    align-items: center;
    -webkit-appearance: none;
    appearance: none;
    background: white;
    border: unset;
    border-radius: var(--radii-0);
    box-sizing: border-box;
    cursor: pointer;
    display: flex;
    flex-shrink: 0;
    justify-content: center;
    padding: unset;
    transition: all 100ms ease-out;
  }

  .c-oUmPa[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-jryFOR {
    font-family: var(--fonts-display);
    font-weight: 700;
    margin: 0;
  }

  .c-dhzjXW {
    display: flex;
  }

  .c-dIzwvf {
    background: white;
    box-shadow: var(--shadows-3);
    box-sizing: border-box;
    left: 50%;
    max-width: 90vw;
    max-height: 90vh;
    overflow-y: auto;
    padding: var(--space-5);
    position: fixed;
    top: 50%;
    transform: translate3d(-50%, -50%, 0);
    z-index: 1147483646;
  }

  .c-dIzwvf:focus {
    outline: none;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-dIzwvf[data-state="open"] {
      animation: k-iaWRFC 550ms cubic-bezier(0.22, 1, 0.36, 1);
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-dIzwvf[data-state="closed"] {
      animation: k-gturUW 550ms cubic-bezier(0.22, 1, 0.36, 1);
    }
}

  .c-fWhrtB {
    background-color: var(--colors-alpha600);
    position: fixed;
    inset: 0;
    z-index: 1147483646;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-fWhrtB[data-state="open"] {
      animation: k-eyOShd 250ms ease-out;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-fWhrtB[data-state="closed"] {
      animation: k-bHbNKp 550ms ease-out;
    }
}
}

@media  {
  .c-oUmPa-PrXKS-size-md {
    height: var(--sizes-4);
    width: var(--sizes-4);
  }

  .c-hMRxhp-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-jryFOR-dtbHih-size-sm {
    font-size: var(--fontSizes-lg);
    line-height: 1.14;
  }

  .c-jryFOR-dtbHih-size-sm::before {
    content: '';
    margin-bottom: -0.2174em;
    display: table;
  }

  .c-jryFOR-dtbHih-size-sm::after {
    content: '';
    margin-top: -0.2254em;
    display: table;
  }

  .c-dhzjXW-hmJoPD-justify-end {
    justify-content: end;
  }

  .c-dhzjXW-dvnNgW-gap-3 {
    gap: var(--space-3);
  }

  .c-dIzwvf-cJSEWp-size-sm {
    width: 480px;
    height: auto;
    max-width: auto;
    max-height: auto;
  }

@supports (height: 100svh) {
    .c-dIzwvf-cJSEWp-size-sm {
      height: auto;
      max-height: auto;
    }
}

  .c-dIzwvf-cJSEWp-size-sm {
    border-radius: var(--radii-1);
  }
}

<div
  aria-describedby="radix-:rh:"
  aria-label="Dialog"
  aria-labelledby="radix-:rg:"
  class="c-dIzwvf c-dIzwvf-cJSEWp-size-sm"
  data-state="open"
  id="radix-:rf:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  CONTENT
</div>
`;
